maxim/dataset/dataset.py

```
@@ -1,12 +1,18 @@
 import json
-from typing import Any, Dict, Optional
-
+from typing import Any, Dict, List, Optional, Union
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)